CorsixTH/Src/run_length_encoder.cpp

```diff
@@ -25,79 +25,79 @@ SOFTWARE.
 #include <new>
 #include <algorithm>
 
-IntegerRunLengthEncoder::IntegerRunLengthEncoder()
+integer_run_length_encoder::integer_run_length_encoder()
```

Comment:
`integer` has no meaning does it?
Also `run_length` could be considered an implementation detail, but that somewhat depends on what other encoders we have.

Reply:
Perhaps should be uint32_run_length_encoder/decoder, because it encodes / decodes uint32s. Being a RLE affects the generated result and means the corresponding decoder has to be compatible (i.e. not an implementation detail). It's a fairly low level concept so I think the name makes sense (except 'integer').